### PR TITLE
Add endpoint that returns forecast for a city

### PR DESCRIPTION
**Pull request by lrs8810**
#3 Forecast for a city 
- Add `api/v1/forecast/location=denver,co` and `api/v1/favorites` routes to `app.js`
- Rename and add forecast and favorite files
- Add and require dotenv, node-fetch
- Add Forecast pojo that formats forecast data
- Happy path working

### CODE REVIEW

File: routes/api/v1/favorites.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+   // fetch ("google?=loc")
+   // .then(loc.info  => {
+        // if response.status return
+        // else
+        // fetch('darksky?=latlong')
+        // .then(darksky.result =>{
+        // response.send(darksky.results)
+      // })
+
```
Comment (ap2322) [inline routes/api/v1/favorites.js:50]: Good idea to grab the latlong from google and store it in db

File: routes/api/v1/favorites.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+
+
+});
+
+// steps
+// 1. validate user
+// 2. get location (lat, long from Geocode)
+// 3.
```
Comment (ap2322) [inline routes/api/v1/favorites.js]: 3. add location to favorites table with user_id
4. return success/error responses

File: routes/api/v1/forecast.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      } else {
+        response.status(404).json({
+          error: `Could not find user with api_key ${request.body.api_key}`
+        })
+      }
+  })
+
+  const location = request.query.location
+
+  fetch(`https://maps.googleapis.com/maps/api/geocode/json?key=${process.env.GOOGLE_API_KEY}&address=${location}`)
```
Comment (ap2322) [inline routes/api/v1/forecast.js]: I really like the logic flow you have here of checking user with validation responses before making the fetch call

File: routes/api/v1/forecast.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    .then(response => response.json())
+    .then(result => {
+      let coords = result.results[0].geometry.location
+      fetch(`https://api.darksky.net/forecast/${process.env.DARKSKY_API_KEY}/${coords.lat},${coords.lng}`)
```
Comment (ap2322) [inline routes/api/v1/forecast.js:38]: great work figuring out `.then` and `fetch`. New things are HARD

File: routes/api/v1/forecast.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        .then(response => response.json())
+        .then(result => {
+          let current = result.currently
+        })
+    })
+    .catch((error) => console.error({ error }))
```
Comment (ap2322) [inline routes/api/v1/forecast.js]: Per review in group, include a `response.status(500).send({error}` in the final catch

**Review by ap2322 — COMMENTED** [review]
Great progress on Forecast and Favorites, almost there!

File: routes/api/v1/forecast.js (hunk 3 above)
Comment (lrs8810) [reply routes/api/v1/forecast.js]: Thanks, Alice!

**Pull request merged**